Observer/BackfillTransactions.php

```diff
@@ -15,309 +15,347 @@
+            ],
+        ];
+
+        $data = [
+            'data' => [
+                'bulk_uuid' => $this->uuid,
+                'topic_name' => \Taxjar\SalesTax\Model\Configuration::TAXJAR_TOPIC_SYNC_TRANSACTIONS,
+                'serialized_data' => $this->serializer->serialize($dataToEncode),
+                'status' => \Magento\Framework\Bulk\OperationInterface::STATUS_TYPE_OPEN,
+            ]
+        ];
+
+        return $this->operationFactory->create($data);
     }
 
     /**
-     * @param $websiteId
-     * @return array
+     * @param \Magento\Store\Api\Data\StoreInterface $store
+     * @return bool
      */
-    protected function getWebsiteStoreIds($websiteId): array
+    private function compareStoreWebsiteId(\Magento\Store\Api\Data\StoreInterface $store): bool
     {
-        $storeIds = [];
-
-        foreach ($this->storeManager->getStores() as $store) {
-            if ($store->getWebsiteId() == $websiteId) {
-                $storeIds[] = $store->getId();
-            }
-        }
+        $websiteId = $store->getWebsiteId();
+        return (!is_null($websiteId) && $websiteId == $this->getInput('websiteId'));
+    }
 
-        return $storeIds;
+    /**
+     * @param \Magento\Sales\Api\Data\OrderInterface $order
+     * @return bool
+     */
+    private function isOrderSyncable(\Magento\Sales\Api\Data\OrderInterface $order): bool
+    {
+        return $order->getUpdatedAt() > $order->getTjSalestaxSyncDate();
     }
 
     /**
-     * @param AbstractModel $object
+     * @param \Magento\Sales\Model\AbstractModel $object
      * @return int|null
      */
-    protected function getEntityId($object): ?int
+    private function getEntityId(\Magento\Sales\Model\AbstractModel $object): ?int
     {
-        return $object->getEntityId();
+        return (int)$object->getEntityId();
     }
 
     /**
-     * @param string $bulkUuid
-     * @param $body
-     * @return OperationInterface
+     * @return array
      */
-    protected function makeOperation(
-        string $bulkUuid,
-        $body
-    ): OperationInterface {
-        $dataToEncode = [
-            'meta_information' => $body,
-        ];
-        $data = [
-            'data' => [
-                'bulk_uuid' => $bulkUuid,
-                'topic_name' => TaxjarConfig::TAXJAR_TOPIC_SYNC_TRANSACTIONS,
-                'serialized_data' => $this->serializer->serialize($dataToEncode),
-                'status' => \Magento\Framework\Bulk\OperationInterface::STATUS_TYPE_OPEN,
-            ]
+    private function getConfiguration(): array
+    {
+        try {
+            $criteria = $this->getSearchCriteria()->__toArray();
+        } catch (\Exception $e) {
+            $criteria = null;
+        }
+
+        return [
+            'from' => $this->getInput('from'),
+            'to' => $this->getInput('to'),
+            'force_sync' => (bool)$this->getInput('force'),
+            'criteria' => $criteria,
```

Comment:
I'm not sure if we need to include the criteria in the logged configuration data. Because we generate the criteria only using the from and to inputs those may be sufficient to replicate any issue that occurred by using the logs.